astra/src/test/java/com/slack/astra/graphApi/GraphBuilderTest.java

```diff
@@ -0,0 +1,610 @@
+package com.slack.astra.graphApi;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.slack.astra.zipkinApi.ZipkinEndpointResponse;
+import com.slack.astra.zipkinApi.ZipkinSpanResponse;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class GraphBuilderTest {
+  private GraphBuilder defaultGraphBuilder;
+  private GraphBuilder configuredGraphBuilder;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    defaultGraphBuilder = new GraphBuilder(GraphConfig.DEFAULT);
+
+    // Load custom config from YAML file
+    Path configPath =
+        new File(
+                getClass()
+                    .getClassLoader()
+                    .getResource("test-dependency-graph-config.yaml")
+                    .getFile())
+            .toPath();
+    GraphConfig customConfig = GraphConfig.load(configPath);
+    configuredGraphBuilder = new GraphBuilder(customConfig);
+  }
+
+  @Test
+  void buildFromSpans_emptyList_returnsEmptyGraph() {
+    List<ZipkinSpanResponse> spans = new ArrayList<>();
+    Graph graph = defaultGraphBuilder.buildFromSpans(spans);
+
+    assertThat(graph.nodes).isEmpty();
+    assertThat(graph.edges).isEmpty();
+  }
+
+  @Test
+  void buildFromSpans_singleSpanWithoutParent_createsSingleNodeWithoutEdges() {
+    List<ZipkinSpanResponse> spans = new ArrayList<>();
+    ZipkinSpanResponse span =
+        createSpanWithTags(
+            "span1",
+            "trace1",
+            null,
+            Map.of(
+                "kube.app",
+                "app1",
+                "kube.namespace",
+                "ns1",
+                "kube.operation",
+                "op1",
+                "resource",
+                "res1"));
+    spans.add(span);
+
+    Graph graph = configuredGraphBuilder.buildFromSpans(spans);
+
+    assertThat(graph.nodes).hasSize(1);
+    Node node = graph.nodes.get(0);
+
+    // Verify the node ID matches the expected hash
+    SortedMap<String, String> expectedMetadata =
+        new TreeMap<>(
+            Map.of(
+                "app", "app1",
+                "namespace", "ns1",
+                "operation", "op1",
+                "resource", "res1"));
+    String expectedId = Node.generateIdFromMetadata(expectedMetadata);
+
+    assertThat(node.getId()).isEqualTo(expectedId);
+    assertThat(node.getMetadata().get("app")).isEqualTo("app1");
+    assertThat(node.getMetadata().get("namespace")).isEqualTo("ns1");
+    assertThat(node.getMetadata().get("operation")).isEqualTo("op1");
+    assertThat(node.getMetadata().get("resource")).isEqualTo("res1");
```

Comment:
you could just assert that getMetadata equals the expected metadata instead of having individual asserts for each